nexus/src/app/background/sync_switch_configuration.rs

```diff
@@ -51,8 +51,8 @@ use omicron_common::{
 use serde_json::json;
 use sled_agent_client::types::{
     BgpConfig as SledBgpConfig, BgpPeerConfig as SledBgpPeerConfig,
-    EarlyNetworkConfig, EarlyNetworkConfigBody, HostPortConfig, PortConfigV1,
-    RackNetworkConfigV1, RouteConfig as SledRouteConfig,
+    EarlyNetworkConfig, EarlyNetworkConfigBody, HostPortConfig, PortConfigV2,
+    RackNetworkConfigV2, RouteConfig as SledRouteConfig, UplinkAddressConfig,
```

Comment:
I'd probably prefer `RackNetworkConfig` here over an explicit version so that any new versions force changes to be mirrored here. Ditto with `PortConfig`. I think this should play nicely with the existing replace directives pointing at `api::internal::shared` -- if not, then it's not a deal-breaker.

Reply:
I should have moved it back to "draft", as I expect this to take a few iterations before it's ready for re-review.

Anyway, thanks for taking a look.  This sounds like a good idea.  I'll give it a try and see how it shakes out.

Reply:
I took a look at this, and I think it would actually have the opposite effect.  Changing the structure without changing the name would make it easy to overlook a change on the client side - especially if we `#[derive(Default)]`.  

I've made the older versions of the structs non-public, which will force any current users to adapt to the new structure.

Reply:
Okay, that makes a lot of sense to me -- I think that's a good way to go about enforcing the correct behaviour. Thanks for the extra time spent here!